source/adapters/level_zero/program.cpp

```diff
@@ -657,19 +659,32 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramGetInfo(
     // device.  Since Level Zero supports only one device, there is only one
     // pointer.  If the pointer is NULL, we don't do anything.  Otherwise, we
```

Comment:
> Since Level Zero supports only one device, there is only one pointer.

This is no longer true. Lower down this function, zeModule is used, which is only going to be set for native handles. Should this use `ZeModuleMap` instead (provided that it exists).

This is yet another situation stemming from duplicate zeModule state.

Reply:
Good point, I will adjust that such that the functionality is retrieving the matched module.

Reply:
Fixed in my latest patch, please take a look at the latest changes @pbalcer , thnaks!